File: c_sharp/BestTimetoBuyandSellStock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int diff = 0;
        int min = prices[0];

        for (int i = 1; i < prices.Length; i++) {
            if (min > prices[i]) {
                min = prices[i];
            } else {
                if (diff < prices[i] - min) {
                    diff = prices[i] - min;
                }
            }
        }

        return diff;
    }
}

File: python/BestTimetoBuyandSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        diff = 0
        m = prices[0]
        for i in range(1, len(prices)):
            m = min(m, prices[i])
            diff = max(diff, prices[i] - m)

        return diff